### PR TITLE
Add test 'Sort and pagination countries table'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add test 'Sort and pagination countries table'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/11_international/02_locations/02_countries/04_sortAndPagination" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21403)

### CODE REVIEW

File: tests/UI/pages/BO/international/locations/countries/index.js [hunk 1]
```diff
@@ -198,6 +211,25 @@ class Countries extends BOBasePage {
...
+    const rowsNumber = await this.getNumberOfElementInGrid(page);
+    const allRowsContentTable = [];
+    for (let i = 1; i <= rowsNumber; i++) {
+      let rowContent = await this.getTextColumnFromTable(page, i, columnName);
+      if (rowContent === '-') {
+        rowContent = '0';
+      }
+      await allRowsContentTable.push(rowContent);
+    }
+    return allRowsContentTable;
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/countries/index.js]:
```suggestion
    const rowsNumber = await this.getNumberOfElementInGrid(page);
    const allRowsContentTable = [];
    
    for (let i = 1; i <= rowsNumber; i++) {
      let rowContent = await this.getTextColumnFromTable(page, i, columnName);
      
      if (rowContent === '-') {
        rowContent = '0';
      }
      
      await allRowsContentTable.push(rowContent);
    }
    
    return allRowsContentTable;
```

File: tests/UI/pages/BO/international/locations/countries/index.js [hunk 2]
```diff
@@ -253,6 +285,88 @@ class Countries extends BOBasePage {
...
+        columnSelector = this.sortColumnDiv(5);
+        break;
+
+      case 'z!id_zone':
+        columnSelector = this.sortColumnDiv(6);
+        break;
+
+      default:
+        throw new Error(`Column ${sortBy} was not found`);
+    }
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/countries/index.js:326]:
```suggestion
    }
    
```

File: tests/UI/pages/BO/international/locations/countries/index.js [hunk 3]
```diff
@@ -253,6 +285,88 @@ class Countries extends BOBasePage {
...
+  /**
+   * Select pagination limit
+   * @param page
+   * @param number
+   * @returns {Promise<string>}
+   */
+  async selectPaginationLimit(page, number) {
+    await this.waitForSelectorAndClick(page, this.paginationDropdownButton);
+    await this.waitForSelectorAndClick(page, this.paginationItems(number));
+    return this.getPaginationLabel(page);
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/countries/index.js]:
```suggestion
    await this.waitForSelectorAndClick(page, this.paginationItems(number));
    
    return this.getPaginationLabel(page);
```

**Review by SimonGrn — DISMISSED** [review]

File: tests/UI/pages/BO/international/locations/countries/index.js [hunk 4]
```diff
@@ -253,6 +285,88 @@ class Countries extends BOBasePage {
...
+  }
+
+  /**
+   * Click on next
+   * @param page
+   * @returns {Promise<string>}
+   */
+  async paginationNext(page) {
+    await this.clickAndWaitForNavigation(page, this.paginationNextLink);
+    return this.getPaginationLabel(page);
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/countries/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(page, this.paginationNextLink);
    
    return this.getPaginationLabel(page);
```

File: tests/UI/pages/BO/international/locations/countries/index.js [hunk 5]
```diff
@@ -253,6 +285,88 @@ class Countries extends BOBasePage {
...
+  }
+
+  /**
+   * Click on previous
+   * @param page
+   * @returns {Promise<string>}
+   */
+  async paginationPrevious(page) {
+    await this.clickAndWaitForNavigation(page, this.paginationPreviousLink);
+    return this.getPaginationLabel(page);
```
Comment (NeOMakinG) [inline tests/UI/pages/BO/international/locations/countries/index.js]:
```suggestion
    await this.clickAndWaitForNavigation(page, this.paginationPreviousLink);
    
    return this.getPaginationLabel(page);
```

**Review by NeOMakinG — COMMENTED** [review]
Some comments in order to improve readability

**Review by NeOMakinG — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Pull request merged**